scripts/debug_tools/reduce.py

```diff
@@ -0,0 +1,304 @@
+import argparse
+import json
+import operator
+import os
+import re
+import shutil
+import stat
+import subprocess
+import sys
+import tempfile
+
+
+def check_files_existence(report_dir):
+    '''
+    This function checks the existence of some files which are required for the
+    reduction process.
+    '''
+    def check(path, isfile):
+        if not (os.path.isfile(path) if isfile else os.path.isdir(path)):
```

Comment:
Couldn't we use `os.path.exists` here: https://docs.python.org/2/library/os.path.html